src/client/api/rpc.c

```diff
@@ -167,6 +167,7 @@ daos_rpc_proto_query(crt_opcode_t base_opc, uint32_t *ver_array, int count, int
 	rproto->array_size = count;
 	rproto->ep.ep_grp = sys->sy_group;
 	rproto->ep.ep_tag = 0;
+	rproto->base_opc = base_opc;
```

Comment:
one comment, on the ticket we noticed that the error propagation was messed up.  Does this cause that or do we have a secondary issue?

Reply:
No, it failed first time as timeout, and then we re-choose a new rank and retry, but the second time, we passed wrong base_opc because base_opc did not init properly, so return DER_NONEXIST is expected IMO.